app/controllers/requests_controller.rb

```diff
@@ -3,19 +3,25 @@
 # crud for tech requests
 class RequestsController < ApplicationController
   def index
+    authorize Request
     @requests = Request.all
   end
 
   def new
+    authorize Request
     @request = Request.new
     @people = Person.all
     @techies = Techie.all
   end
 
   def create
+    authorize Request
     @request = Request.new(request_params)
-    redirect_to :requests, notice: "Request created successfully" if @request.save
-    flash[:alert] = "There was an issue creating your request."
+    if @request.save
+      redirect_to :requests, notice: 'Request created successfully'
+    else
+      flash[:alert] = 'There was an issue creating your request.'
```

Comment:
Is there a corresponding 'create' partial? I can't find one, so I think this unhappy path will fail. The typical pattern is to render the action's partial that initiated the request during an unhappy mutation action, so 'new' in this case.

If we do that, you will also need to change 'flash' to 'flash.now'. It is probably instructive to see why you want this change to flash—i only understood after seeing one too many flash messages persisting after I navigated the page away.